README.md

```diff
@@ -77,10 +77,65 @@ expect("foobar").toStartWith("foo");
 // Number assertion
 expect(sum(1, 2)).toBePositive();
 
+// Error assertion
+expect(new Error(errorMessage)).toHaveMessage(expectedError);
+
+// Array assertion
+const data = [1, 2, 3, 4]
+expect(data).toMatchAll(x => x < 5);
+expect(data).toBeEmpty()
+
+// Date assertion
+const date = new Date(2023, 12, 31);
+expect(date).toBeAfter(new Date(2023, 12, 1));
+expect(date).toBeBefore(new Date(2024, 1, 1));
+
+// Object assertion
+const objectData = {
+  key1: "test1",
+  key2: "test2",
+};
+expect(objectData).toContainKey("key1");
+expect(objectData).toContainEntry(["key1", "test1"]);
+
 expect(14).toEndWith("4");
            ^ ? type error: `toEndWith` does not exist in `NumberAssertion`
 ```
 
+You can also assert from synchronous and asynchronous code, for example:
+
+```ts
+const verifyEnvVar = (): void => {
+  const { MY_ENV_VAR } = process.env;
+
+  if (!MY_ENV_VAR) {
+    throw new Error("Missing MY_ENV_VAR environment variable");
+  }
+};
```

Comment:
Top-level functions are usually better as named function declarations instead of arrow function expressions because of [hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting) of the declaration and the scope of the `this` value. Here's a great 3mins read that dives deep into this: https://kentcdodds.com/blog/function-forms

In this case, were not running this code, but we should still keep good practices and conventions 🙂 
```suggestion
function verifyEnvVar(): void {
  const { MY_ENV_VAR } = process.env;

  if (!MY_ENV_VAR) {
    throw new Error("Missing MY_ENV_VAR environment variable");
  }
}
```